api/hypershift/v1beta1/azure_types.go

```diff
@@ -0,0 +1,272 @@
+type AzureNodePoolPlatform struct {
+	// vmSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	// The size naming convention is documented here https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions.
+	// Size names should start with a Family name, which is represented by one of more capital letters, and then be followed by the CPU count.
+	// This is followed by 0 or more additional features, represented by a, b, d, i, l, m, p, t, s, C, and NP, refer to the Azure documentation for an explanation of these features.
+	// Optionally an accelerator such as a GPU can be added, prefixed by an underscore, for example A100, H100 or MI300X.
+	// The size may also be versioned, in which case it should be suffixed with _v<version> where the version is a number.
+	// For example, "D32ads_v5" would be a suitable general purpose VM size, or "ND96_MI300X_v5" would represent a GPU accelerated VM.
+	//
+	// +kubebuilder:validation:Pattern=`^(Standard_|Basic_)?[A-Z]+[0-9]+(-[0-9]+)?[abdilmptsCNP]*(_[A-Z]*[0-9]+[A-Z]*)?(_v[0-9]+)?$`
+	// +kubebuilder:validation:Required
+	// + Azure VM size format described in https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
+	// + "[A-Z]+[0-9]+(-[0-9]+)?" - Series, size and constrained CPU size
+	// + "[abdilmptsCNP]*" - Additive features
+	// + "(_[A-Z]*[0-9]+[A-Z]*)?" - Optional accelerator types
+	VMSize string `json:"vmSize"`
+
+	// image is used to configure the VM boot image. If unset, the default image at the location below will be used and
+	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
+	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
+	// Hosted Cluster specification respectively, HostedCluster.Spec.Platform.Azure.SubscriptionID and
+	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="image is immutable"
+	// +kubebuilder:validation:Required
+	Image AzureVMImage `json:"image"`
+
+	// osDisk provides configuration for the OS disk for the nodepool.
+	// This can be used to configure the size, storage account type, encryption options and whether the disk is persistent or ephemeral.
+	// When not provided, the platform will choose reasonable defaults which are subject to change over time.
+	// Review the fields within the osDisk for more details.
+	OSDisk AzureNodePoolOSDisk `json:"osDisk"`
+
+	// availabilityZone is the failure domain identifier where the VM should be attached to.
+	// This must not be specified for clusters in a location that does not support AvailabilityZone because... TODO: why?
+	// Example values are TODO
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
```

Comment:
IIUC correctly, AZs in azure are actually integers, but we represent them as strings? Typically `"1"`, `"2"` or `"3"`